src/vs/workbench/services/hover/browser/hoverService.ts

```diff
@@ -31,23 +32,26 @@ export class HoverService implements IHoverService {
 		@IInstantiationService private readonly _instantiationService: IInstantiationService,
 		@IContextViewService private readonly _contextViewService: IContextViewService,
 		@IContextMenuService contextMenuService: IContextMenuService,
-		@IKeybindingService private readonly _keybindingService: IKeybindingService
+		@IKeybindingService private readonly _keybindingService: IKeybindingService,
+		@IAccessibilityService private readonly _accessibilityService: IAccessibilityService
 	) {
 		contextMenuService.onDidShowContextMenu(() => this.hideHover());
 	}
 
-	showHover(options: IHoverOptions, focus?: boolean): IHoverWidget | undefined {
+	showHover(options: IHoverOptions, focus?: boolean, skipLastFocusedUpdate?: boolean): IHoverWidget | undefined {
 		if (getHoverOptionsIdentity(this._currentHoverOptions) === getHoverOptionsIdentity(options)) {
 			return undefined;
 		}
 		this._currentHoverOptions = options;
 		this._lastHoverOptions = options;
-		if (options.trapFocus && document.activeElement) {
-			this._lastFocusedElementBeforeOpen = document.activeElement as HTMLElement;
-		} else {
-			this._lastFocusedElementBeforeOpen = undefined;
+		options.trapFocus = options.trapFocus || this._accessibilityService.isScreenReaderOptimized();
+		if (!skipLastFocusedUpdate) {
```

Comment:
Let's add `// HACK: ...` above this line to make it obvious we don't want this here forever